src/share/poudriere/common.sh

```diff
@@ -3803,17 +3802,20 @@ download_hybridset_pkg_from_repo() {
 
 	msg "Bootstrapping hybrid ABI pkg."
 
-	hybridset_pkgcmd "${arch}" "${MASTERMNT}" "/" install -q pkg
-	hybridset_pkgcmd "${arch}" "${MASTERMNT}" "/" update -q
+	hybridset_pkgcmd install -q pkg
+	if [ $? -ne 0 ]; then
+		err 1 "Failed to install hybrid ABI pkg"
+	fi
+	hybridset_pkgcmd update -q
+	if [ $? -ne 0 ]; then
+		err 1 "Failed to update hybrid ABI repository"
+	fi
 
-	get_host_abi host_abi
-	if [ "${host_abi}" = "purecap" ]; then
-		pkgcmd="pkg64"
-	else
-		pkgcmd="pkg"
+	if ! hybridset_is_really_host; then
+		return
 	fi
```

Comment:
Why are you returning here? At the beginning of this function, we return in non-CheriABI cases. This check makes the rest of this function ignored as it returns in the CheriABI case.

We need to update /usr/local64/etc/pkg.conf to set `ABI` to `FreeBSD:14:{amd64,arm64}` when building on FreeBSD.

Reply:
(hybridset_is_really_host is supposed to abstract away precisely whether you want to mess with pkg.conf to be host packages or not, that's the whole point of it)

Reply:
Isn't the rest of this function just dead code?

Reply:
Also, `hybridset_is_really_host` is a really confusing name. Maybe `hybridset_is_native` would be more informative.